src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -751,12 +752,38 @@ void MetaspaceShared::preload_classes(TRAPS) {
     }
   }
 
-  // Exercise the manifest processing code to ensure classes used by CDS at runtime
-  // are always archived
+  // Some classes are used at CDS runtime but are not loaded, and therefore archived, at
+  // dumptime. We can perform dummmy calls to these classes at dumptime to ensure they
+  // are archived.
```

Comment:
This code doesn't just load classes though, it loads, links, initializes them and creates a unwanted instance. Isn't there a simpler way to do this? Do these classes have to be initialized?

Reply:
The classes make calls to other classes that need to be loaded as well, this code acts like a shortcut to load a list of classes that we use at runtime. These calls mirror the ones used at runtime.